### ISSUE TITLE
Questions about the data setup and how to setup my own datasets?

### ISSUE THREAD

**Issue by RodneyPerhaps** [Brummi/derender3d#1]
**Amazing work there!** Very glad you release the training code and the preprocessed Co3D datasets. 

But I cannot reproduce the Co3D dataset or the CelebA-HQ dataset. 

What is the “**extracted**” folder in the dataloader?
<img width="540" alt="train val precomputed dir  cfas 9et(train val precomputed dir'  None)" src="https://user-images.githubusercontent.com/78194960/170930502-231cb307-6ea5-4fb3-9abe-cdaf56d97f4b.png">

Why are the dimensions of light (1x3) and view matrix (2x3) **different** from those of unsup3d?

When will the **data setup scripts** be **released**?  

How could I do the preprocessing step on **my own** or other published scanned object datasets such as **GSO** (Google Scanned Objects)? 

Thanks again for this fantastic job!

**Comment by Brummi** [comment]
Hi Rodney,

thanks for showing interest in our work!
The differentiation between extracted and precomputed is somewhat of a legacy thing. We started out by experimenting with data we got from Unsup3D -> precomputed coarse shape, albedo, etc.
However, then we wanted to restrict ourselves to shape only and extracted albedo and light ourselves from just the coarse shape and the input image. (Motivation: Coarse shape is easy to get form many different methods, coarse albedo and light is not).

The dimensions of the light in the data should be either 1x4 [ambient strength, directional strength, d_x, d_y] (d_z would then be assumed 1), or 1x5 [ambient strength, directional strength, d_x, d_y, d_z].

The view can be parameterized by 1x6 [r_x, r_y, r_z, t_x, t_y, t_z] (Unsup3D), or 3x3 rotation matrix (CO3D). Note, however, that the view is also a relic from when we used the precomputed light from Unsup3D. When extracting albedo and light ourselves, the view can always be considered to be irrelevant.

Data setup script: We are working on them right now, so it should be within the next few weeks. However, we can't guarantee that it will be before CVPR.

Best,
Felix

**Comment by RodneyPerhaps** [comment]
Hello Felix,

Thank you very much for such a detailed answer! BTW, have you considered adding face parsing to the face domain training pipeline and calculating only the loss of the foreground face? After all, adding the loss of the background only deteriorates the network performance.

**Comment by Brummi** [comment]
Hi Rodney,

actually, we haven't tried this out yet. (Similarly to Unsup3D, we use very tight crops around the face). 
But you are right, this would be an interesting direction for future experiments. One of the initial reaons for the specularity refinement step was to avoid weird specularity on hair, which has a very different material from the face. Simply masking away the hair would remove these issues.

Best,
Felix

**Comment by RodneyPerhaps** [comment]
So it's worth a try. Many thanks!
